### ISSUE TITLE
React example could need a ES6 version

### ISSUE THREAD

**Issue by iehdk** [VeliovGroup/Meteor-Files#572]
A ES6 version of the React Example would be most welcome, especially, to demonstrate how to replace the mixin with a HOC.

**Comment by dr-dimitru** [comment]
Hello @iehdk ,

Thank you for suggestion. PRs are  always welcome :)
 - Doc's source is here - https://github.com/VeliovGroup/Meteor-Files/blob/master/docs/react-example.md

**Comment by iehdk** [comment]
Sure, I am not sure how to put this together. I think ReactMeteorData is best used with a Wrapper component. I did have this working for FS.Collection (which I want out of because of utterly confusing docs and examples and lack of activity). I suppose it just need `meteor-files` flavoring.

`imports/api/files.collection.js`:

```
/* globals Meteor FS */

// FS.debug = true;
import MAX_FILE_SIZE from '../../imports/startup/both/constants';

const Files = new FS.Collection('files', {
  stores: [new FS.Store.FileSystem('files', { path: '~/scratch/uploads' })],
  filter: {
    maxSize: MAX_FILE_SIZE, // in bytes
    allow: {
      contentTypes: ['text/csv', 'text/plain'],
      extensions: ['csv', 'txt'],
    },
    onInvalid: (message) => {
      if (Meteor.isClient) {
        alert(message); // FIXME Replace with modal alert
      } else {
        console.log(message);
      }
    },
  },
});

if (Meteor.isServer) {
  Files.allow({
    insert: () => { return true; },
    download: () => { return true; },
    remove: () => { return true; },
  });

  Meteor.publish('allFiles', () => {
    return Files.find();
  });

  Meteor.publish('lastFile', () => {
    return Files.find({}, { sort: { updatedAt: -1 }, limit: 1 });
  });

  Meteor.publish('file', () => {
    return Files.find({ _id: this._id });
  });
}

export default Files;
```


And the wrapper component:

```

/* globals Meteor HTTP */

import { withTracker } from 'meteor/react-meteor-data';
import NavBar from '../../client/components/navbar.component';
import Files from '../../imports/api/files.collection';

const fileSubscription = () => {
  const filesSub = Meteor.subscribe('allFiles');

  return {
    filesReady: filesSub.ready(),
    fileUpload: (file, callback) => {
      Files.insert(file, callback);
    },
    fileDownload: (url, callback) => {
      HTTP.get(url, callback); // Missing upload complete guarantee
    },
    files: Files.find({}, {
      sort: { uploadedAt: -1 }, // FIXME
    }).fetch(),
  };
};

export default NavBarContainer = withTracker(fileSubscription)(NavBar);
```

**Comment by mikkelking** [comment]
Did you resolve this? I managed to get it working, with a little bit of wrangling.

If you like I could create a small meteor project that you could use as an example. Just let me know

**Comment by dr-dimitru** [comment]
Closed by #600 in [`v1.9.10`](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.9.10).
Thanks to @mikkelking

Feel free to reopen it in case if the issue is still persists on your end.
